### ISSUE TITLE
project domain name

### ISSUE THREAD

**Issue by fricklerhandwerk** [nix-open-org/org-v2#80]
we observe that people are confused about names in the Nix ecosystem and find it hard to distinguish between its components.

given the goal of easing learning Nix, we should reduce that confusion as much as possible.

proposal:
- match domain names to component names

considerations:
  - Nix and Nixpkgs are not used separately at large
    - its possible though and some do it
  - NixOS covers a distinct use case, and only ~half of the [2022 survey](https://discourse.nixos.org/t/2022-nix-survey-results/18983) respondents use NixOS daily
    - @MMesch reports that most business users are not on NixOS
  - Nix is at the core of the ecosystem, everything builds on top and around it 

current state:
- [nixos.org](https://nixos.org/) used for everything
- [nixpkgs.org](https://nixpkgs.org/) redirects to [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs)
- [nix.store](https://nix.store) currently [redirects to nixos.org](https://discourse.nixos.org/t/nix-related-domains-that-i-control/10034/7)
- ~~[nix.dev](https://nix.dev) is a separate project owned by @domenkozar, contents used on [nixos.org/learn](https://nixos.org/learn) [without attribution](https://github.com/NixOS/nixos-homepage/issues/671#issuecomment-791353841)~~
  [nix.dev](https://nix.dev) is an official documentation resource
- [nix.org](https://nix.org) is [currently being squatted on](https://github.com/NixOS/nixos-homepage/issues/633#issuecomment-841634501)

options:
- use [nixpkgs.org](http://nixpkgs.org/) for Nix+Nixpkgs (current [web site](https://github.com/NixOS/nixos-homepage/tree/5112cf6ea6ed7df34e70b8e0a386dcfda8fe0bc6)), redirect [nixos.org](http://nixos.org/) to [NixOS/nixpkgs/tree/master/nixos](https://github.com/NixOS/nixpkgs/tree/master/nixos)
  - contributors more focused on NixOS could set up something more appropriate on their own
- use nix.dev
  - with support from @domenkozar
  - done, but currently only for the purpose of serving a documentation resource
- buy nix.org
  - with support from Foundation (@edolstra @domenkozar @thufschmitt @grahamc)
- [more domain names](https://discourse.nixos.org/t/nix-related-domains-that-i-control/10034)

related:
- NixOS/nix.dev#290
- https://github.com/NixOS/nixos-homepage/issues/633

@garbas this is a responsibility of the marking team, but intimately concerns documentation w.r.t. consistency and avoiding confusion

**Comment by asymmetric** [comment]
On buying nix.org, I think a good (and way cheaper) alternative could be buying nix-lang.org. It's what [many](https://www.rust-lang.org/) [other](https://elixir-lang.org/) languages do.

**Comment by fricklerhandwerk** [comment]
The Nix ecosystem and the Nix package manager are not really about the Nix language (just as Bazel is not really about Starlark). nix-lang.org would easily make the impression the project is somehow revolving around the configuration language. That would defeat the purpose of reducing confusion.

**Comment by asymmetric** [comment]
I disagree. Part of the confusion comes from using the same name for the language and the ecosystem, so renaming the language to something like "nix lang" could reduce confusion. Then, controlling the domain is simply a good idea, whatever we decide to do with it. And potentially, it could be a place that hosts the nix-lang manual (or redirect to it)

**Comment by fricklerhandwerk** [comment]
> Part of the confusion comes from using the same name for the language and the ecosystem

This is true and we should address it with NixOS/nix.dev#275.

> controlling the domain is simply a good idea, whatever we decide to do with it.

Sure, it just does not help with the issue here.

**Comment by davidak** [comment]
i think it would be best to have [nix.org](https://nix.org/) for the package manager + resources about nixpkgs and nix lang and  [nixos.org](https://nixos.org/) for the linux distribution + deployment tools for it, like nixops

that would solve confusion like reported in https://github.com/NixOS/nixos-homepage/issues/832

**Comment by fricklerhandwerk** [comment]
marketing team and Foundation board should take over: https://github.com/NixOS/nix.dev/issues/285

not closing, issue can be migrated.

**Comment by fricklerhandwerk** [comment]
another argument for having a Nix-package-manager-specific web site is to manage the web rendering of Nix manuals in the [Nix repo](https://github.com/NixOS/nix). Right now the [path-level redirects are on `nixos-homepage`](https://github.com/NixOS/nixos-homepage/blob/8e6a0d831113bfd7f1c36788eae871201d09fc18/netlify.toml#L109-L149) while the [anchor-level redirects are on `nix`](https://github.com/NixOS/nix/blob/075bf6e5565aff9fba0ea02f3333c82adf4dccee/doc/manual/redirects.js#L3-L331). redirects are important to keep existing links alive.

Here are two pull requests that should be in sync, but produce a tail of follow-up work that is highly likely to be missed unless someone remembers to update the redirects on nixos.org once the current master branch reaches a release:
- https://github.com/NixOS/nix/pull/6863
- https://github.com/NixOS/nixos-homepage/pull/881

Having the Nix web site in the source repo would allow managing redirects without additional effort.

**Comment by Ericson2314** [comment]
Yeah I want to second that having "NixOS" be the umbrella term for the whole ecosystem is confusing and we should move away from doing that.

**Comment by fricklerhandwerk** [comment]
@edolstra this is for the board to answer, could you please transfer it to NixOS/nixos-foundation?

**Comment by fricklerhandwerk** [comment]
We could even consider [renaming the GitHub organization](https://docs.github.com/en/organizations/managing-organization-settings/renaming-an-organization) to _Nix_ to match the [hopefully-soon-to-be authoritative terminology](https://github.com/NixOS/nix.dev/pull/263) and better support the (technically and morally correct) narrative that Nix is at the core of the Nix ecosystem. I'd even go as far as proposing to rename the NixOS Foundation to Nix Foundation.

**Comment by fricklerhandwerk** [comment]
@domenkozar please transfer this issue to https://github.com/NixOS/nixos-foundation/

**Comment by zimbatm** [comment]
Somebody already registered nixlang.org and nix-lang.org.

I'm holding nixlang.com for the community

**Comment by bryanhonof** [comment]
I recently bought the getnix.org domain. I intend to make this a place to define a few streams, like https://getfedora.org/ does. So, there'd be a little box that explains and links to relevant resources for Nix as a package manager, another one for NixPkgs, one for NixOS, and so forth.

Furthermore, I'm open to transferring this to the NixOS foundation if they ever desire the domain.

**Comment by algorithmiker** [comment]
I noticed `nix.guide` is currently free.
Maybe we can arrange to move nix.dev over to nix.guide and make `nix.dev` the main domain

**Comment by infinisil** [comment]
Moving this to the org repo, as this is about project and technical direction, which doesn't fall under the foundation's purview, but rather the [SC's](https://github.com/NixOS/org/blob/main/doc/constitution.md#detailed-responsibilities).

**Comment by mightyiam** [comment]
Please initiate negotiation for nix.org 🙏

**Comment by Malix-Labs** [comment]
https://github.com/NixOS/nix.dev/issues/1231#issuecomment-4814788009 :
> > "nix" by itself should only ever be used to refer to the entire ecosystem (or the command line tool, to be stylised as `nix`), and that everywhere else NixOS/nixpkgs/nixlang should be used.
> 
> I strongly agree with this, albeit not sure about the capitalization (thinking about it, and since I like the capitatalization of "NixOS", why not "**NixPkgs**" and "**NixLang**" ?)
> 
> Also, I don't think "NixLang" was ever recognize as potentially official prior to this post, even if we probably all used it at some point to clarify that we were talking about the DSL

**Comment by mweinelt** [comment]
> Please initiate negotiation for nix.org 🙏

The trademark is for the term "NIXOS".

**Comment by mightyiam** [comment]
> > Please initiate negotiation for nix.org 🙏
> 
> The trademark is for the term "NIXOS".

See https://github.com/NixOS/nix.dev/issues/1231
